test/e2e/subscription_e2e_test.go

```diff
@@ -1366,6 +1366,409 @@ var _ = Describe("Subscription", func() {
+				Expect(err).ToNot(HaveOccurred())
+			})
+			AfterEach(func() {
+				if catsrcCleanup1 != nil {
+					catsrcCleanup1()
+				}
+				if catsrcCleanup2 != nil {
+					catsrcCleanup2()
+				}
+				if catsrcCleanup3 != nil {
+					catsrcCleanup3()
+				}
+
+			})
+
+			When("creating subscription for the main package", func() {
+				var subscription *v1alpha1.Subscription
+				BeforeEach(func() {
+					// Create a subscription for packageA in catsrc
+					subscriptionSpec := &v1alpha1.SubscriptionSpec{
+						CatalogSource:          catsrcMain.GetName(),
+						CatalogSourceNamespace: catsrcMain.GetNamespace(),
+						Package:                packageMain.PackageName,
+						Channel:                stableChannel,
+						InstallPlanApproval:    v1alpha1.ApprovalAutomatic,
+					}
+					subscriptionName := genName("sub-")
+					cleanupSubscription = createSubscriptionForCatalogWithSpec(GinkgoT(), crClient, testNamespace, subscriptionName, subscriptionSpec)
+
+					var err error
+					subscription, err = fetchSubscription(crClient, testNamespace, subscriptionName, subscriptionHasInstallPlanChecker)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(subscription).ToNot(BeNil())
+
+					_, err = fetchCSV(crClient, mainCSVName, testNamespace, csvSucceededChecker)
+					Expect(err).ToNot(HaveOccurred())
+
+				})
+				AfterEach(func() {
+					if cleanupSubscription != nil {
+						cleanupSubscription()
+					}
+					if cleanup != nil {
+						cleanup()
+					}
+				})
+				It("choose the dependent package from the same catsrc as the installing operator", func() {
+					// ensure correct CSVs were picked
+					Eventually(func() ([]string, error) {
+						ip, err := crClient.OperatorsV1alpha1().InstallPlans(testNamespace).Get(context.TODO(), subscription.Status.InstallPlanRef.Name, metav1.GetOptions{})
+						if err != nil {
+							return nil, err
+						}
+						return ip.Spec.ClusterServiceVersionNames, nil
+					}).Should(ConsistOf(mainCSVName, rightCSVName))
+				})
+
+			})
+
+		})
+
+		Context("creating CatalogSources (the same dependency exists) with different namespaces", func() {
+			var catsrcCleanup1, catsrcCleanup2, catsrcCleanup3 cleanupFunc
+
+			BeforeEach(func() {
+
+				packageDepRight = registry.PackageManifest{PackageName: "PackageDependent"}
+				csv := newCSV(rightCSVName, testNamespace, "", semver.MustParse("0.1.0"),
+					[]apiextensions.CustomResourceDefinition{crd}, nil, nil)
+				packageDepRight.DefaultChannelName = alphaChannel
+				packageDepRight.Channels = []registry.PackageChannel{{Name: alphaChannel, CurrentCSVName: csv.GetName()}}
+				csvsRight = append(csvsRight, csv)
+
+				csv.Name = wrongCSVName
+				packageDepWrong = packageDepRight
+				packageDepWrong.Channels = []registry.PackageChannel{{Name: alphaChannel, CurrentCSVName: csv.GetName()}}
+				csvsWrong = append(csvsWrong, csv)
+
+				dNS, err := kubeClient.KubernetesInterface().CoreV1().Namespaces().Create(context.TODO(),
```

Comment:
I think this test is intending to check for namespace ordering in the sort, right?

If that's the case, the catalog in this generated namespace won't even be in the set that's being sorted. The only namespaces that are considered for sorting are: 1. the namespace where the subscription is and 2. the namespace that is configured for global catalogs

Reply:
mb, I forgot about this.

Reply:
definately need to have a fix next week